### PR TITLE
add ims image copy

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add ims image copy
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format,
  will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openstack/ims/v1/imagecopy/requests.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+// CrossRegionCopyOpts represents options used to create an image.
+type CrossRegionCopyOpts struct {
+	// the name of the copy image
+	Name string `json:"name" required:"true"`
+	// description of the copy image
+	Description string `json:"description,omitempty"`
+	// the target region name.
+	Region string `json:"region,omitempty" required:"true"`
```
Comment (ShiChangkuo) [inline openstack/ims/v1/imagecopy/requests.go:52]:
```suggestion
	TargetRegion string `json:"region" required:"true"`
```

please sortting the parameters in oder of required and optional

File: openstack/ims/v1/imagecopy/requests.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	// the name of the project in the destination region.
+	ProjectName string `json:"project_name" required:"true"`
```
Comment (ShiChangkuo) [inline openstack/ims/v1/imagecopy/requests.go:54]:
```suggestion
	TargetProjectName string `json:"project_name" required:"true"`
```

File: openstack/ims/v1/imagecopy/requests.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	// the agency name.
+	AgencyName string `json:"agency_name" required:"true"`
+	// the ID of the vault
+	VaultId string `json:"vault_id,omitempty"`
+	// Enterprise project ID
+	EnterpriseProjectID string `json:"enterprise_project_id,omitempty"`
```
Comment (ShiChangkuo) [inline openstack/ims/v1/imagecopy/requests.go:60]: according to the API doc, `enterprise_project_id` is not supportted

File: openstack/ims/v1/imagecopy/requests.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+// CrossRegionCopyImage implements create image request.
+func CrossRegionCopyImage(client *golangsdk.ServiceClient, imageId string, opts CrossRegionCopyOptsBuilder) (r JobResult) {
+	b, err := opts.ToCrossRegionCopyMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+
+	_, r.Err = client.Post(copyCrossRegionURL(client, imageId), b, &r.Body, &golangsdk.RequestOpts{OkCodes: []int{200}})
```
Comment (ShiChangkuo) [inline openstack/ims/v1/imagecopy/requests.go:75]:
```suggestion
	_, r.Err = client.Post(copyCrossRegionURL(client, imageId), b, &r.Body, nil)
```

use the default OkCodes is fine

File: openstack/ims/v1/imagecopy/requests.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	// Enterprise project ID
+	EnterpriseProjectID string `json:"enterprise_project_id,omitempty"`
+}
+
+func (opts CrossRegionCopyOpts) ToCrossRegionCopyMap() (map[string]interface{}, error) {
+	return golangsdk.BuildRequestBody(opts, "")
+}
+
+// CrossRegionCopyImage implements create image request.
+func CrossRegionCopyImage(client *golangsdk.ServiceClient, imageId string, opts CrossRegionCopyOptsBuilder) (r JobResult) {
```
Comment (ShiChangkuo) [inline openstack/ims/v1/imagecopy/requests.go:68]: as the package name is `imagecopy`, it's better to define function names not include CopyImage, so I think the function should be **CrossRegion** or **CrossRegionCopy**

File: openstack/ims/v1/imagecopy/requests.go (hunk 1 above)
Comment (houpeng80) [reply openstack/ims/v1/imagecopy/requests.go]: done

File: openstack/ims/v1/imagecopy/requests.go (hunk 2 above)
Comment (houpeng80) [reply openstack/ims/v1/imagecopy/requests.go]: done

File: openstack/ims/v1/imagecopy/requests.go (hunk 4 above)
Comment (houpeng80) [reply openstack/ims/v1/imagecopy/requests.go]: done

File: openstack/ims/v1/imagecopy/requests.go (hunk 5 above)
Comment (houpeng80) [reply openstack/ims/v1/imagecopy/requests.go]: it has been changed to CrossRegionCopy

**Pull request merged**